src/xdist/scheduler/load.py

```diff
@@ -62,6 +63,9 @@ def __init__(self, config: pytest.Config, log: Producer | None = None) -> None:
         self.node2collection: dict[WorkerController, list[str]] = {}
         self.node2pending: dict[WorkerController, list[int]] = {}
         self.pending: list[int] = []
+        self.do_resched: bool = False
```

Comment:
Can we avoid modifying the vanilla schedulers?
e.g. checking for member definition in `Dession`, e.g. `hasattr(self.sched, 'do_resched')` there

Reply:
Let me see if that works! I'd also like to not modify protocol/existing schedulers